lib/puppet/type/tomcat_install.rb

```diff
@@ -0,0 +1,50 @@
+Puppet::Type.newtype(:tomcat_install) do
+  desc "Installs the software into the given directory."
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
+  autorequire(:user) do
+    self[:user]
+  end
+
+  autorequire(:group) do
+    self[:group]
+  end
+
+  newparam(:catalina_home, namevar: true) do
+    desc "The directory of the Tomcat installation."
+
+    validate do |value|
+      raise ArgumentError, "Catalina home is required." if value.nil?
+    end
+  end
+
+  newparam(:version) do
+    desc "Specifies the owner of the source installation directory."
+
+    validate do |value|
+      raise ArgumentError, "Tomcat version is required." if value !~ /\d{1,2}\.\d{1,2}\.\d{1,2}/
+    end
+  end
+
+  newparam(:user) do
+    desc "Specifies the owner of the source installation directory."
+
+    defaultto 'tomcat'
```

Comment:
Currently the install is extracted as root by default to avoid being owned by the user that the instance runs as (for security reasons). I'm not entirely sure that root is valid, but wanted to ask anyway; what do you think?

Reply:
I am not a security guy, but wouldn't we want the install to be owned by the user precisely for security reasons?

I assume if the instance can be run as root, in theory, you could get root to run commands on the host if there is a security leak.